src/modules/Apps.js

```diff
@@ -0,0 +1,142 @@
+/** ****************************************
+ * Project: leaderboard
+ * File: Apps.js
+ * Created: 3/15/23
+ * Author: Abdullah Al Mamun <mamun1214@gmail.com>
+ ****************************************** */
+import Notification from "./Notification";
+
+class Apps {
+  #_notify;
+
+  constructor() {
+    this.items = [];
+    this.controller = {};
+    this.baseUrl = "https://us-central1-js-capstone-backend.cloudfunctions.net/api/";
+    this.#events();
+    this.#_notify = new Notification;
+  }
+
+  set id($val) {
+    this._id = $val;
+  }
+
+  set success($msg) {
+    this.#_notify.show($msg, 'success');
+  }
+
+  set error($msg) {
+    this.#_notify.show($msg, 'error');
+  }
+
+  get id() {
+    if (typeof this._id === 'undefined') {
+      if (this.getItem('ID').length === 0) {
+        return this.getId();
+      }
+      return this.getItem('ID');
+    }
+    return this._id;
+  };
+
+  #events = () => {
+    const $form = document.querySelector('form');
+    $form.addEventListener('submit', (e) => {
+      e.preventDefault();
+      const $formData = new FormData($form);
+      this.saveScore($formData);
+    });
+
+    const $refresh = document.getElementById('refresh');
+    $refresh.addEventListener('click', (e) => {
+      e.preventDefault();
+      this.getScore();
+    });
+  };
+
+  getId = () => {
+    const $url = this.baseUrl + "games/";
+    const $inputs = { name: 'Leaderboard Game' };
+    const $response = this.#httpRequest($url, 'POST', $inputs);
+    $response.then((res) => {
+      const $id = res.result.split(" ");
+      this.id = $id[3];
+      this.#saveStorage('ID', this.id);
+      return this.id;
+    }).catch((error) => {
+      throw new Error(error);
+    });
+  };
+
+  #addItem = ($item) => {
+    const $ul = document.createElement('ul');
+    $ul.id = 'items';
+    document.querySelector('.score-list').innerHTML = '';
+    $item.forEach((item) => {
+      const $li = document.createElement('li');
+      $li.textContent = `${item.user}: ${item.score}`;
+      $ul.appendChild($li);
+    });
+    document.querySelector('.score-list').appendChild($ul);
+  };
+
+  getScore = () => {
+    const $id = this.id;
+    const $url = this.baseUrl + `games/${$id}/scores/`;
+    const $response = this.#httpRequest($url, 'GET');
+    $response.then((res) => {
+      this.#addItem(res.result);
+    }).catch((error) => {
+      throw new Error(error);
+    });
+  };
```

Comment:
@AbayomiOlaoye You need to run the project on your PC. Project requirement does not require 'dist' folder to push on repo

Reply:
@dev-mamun , I'm not sure you have watched my screen recording. Kindly watch it to see that it was from my pc. the major need for change is that users can't retrieve their inputs. Kindly look into it

Reply:
@AbayomiOlaoye yes i see your video. you did not follow readme instruction and try to run from src folder

Reply:
@dev-mamun, I am a bit disappointed by your manner of approach. You seem threatened by the `request for changes` but really, it shouldn't be. Rather than maintaining a stance of correctness or infallibility, I think it would be more impactful and helpful if you ask questions. As we write codes for machines, we must strive to communicate effectively with fellow humans too.
- Since you are bringing that up now, none of the `npm ...` works on your code. The only way to access it was via liveserver. Here is a screenshot:

file:///home/mindprep/Pictures/Screenshots/Screenshot%20from%202023-03-15%2015-33-31.png

**_Please note that code reviewers are not blood hounds or task masters, we are simply your coding partners whose soul aim is to ensure that you write accessible and working codes, which in turn will impact for career as a developer in the long run._**

Reply:
@dev-mamun, you can ask questions if you have one, regarding this.

Reply:
![image](https://user-images.githubusercontent.com/3351180/225346468-f94fd7d4-fac2-42fc-941e-e0ca6a14120e.png)
i think you have node.js issue because that issue on node modules.